src/value.c

```diff
@@ -67,12 +80,9 @@ SIValue SI_Clone(SIValue v) {
 }
 
 SIValue SI_ShallowCopy(SIValue v) {
-  if (v.type & SI_STRING) {
-    // Re-use the pointer of the input's string value
-    return SI_ConstStringVal(v.stringval);
-  }
-  SIValue dup;
-  memcpy(&dup, &v, sizeof(SIValue));
+  SIValue dup = v;
+  // If the original value owns an allocation, mark that the duplicate shares it
+  if (v.allocation == M_SELF) dup.allocation = M_CONST;
```

Comment:
if think this applies to SIValues of type pointer, worth adding that check before checking the allocation type